shell.nix

```
@@ -1,4 +1,9 @@
-let
-  mypkgs = import ./. {};
-  shellFrom = import ./nix/shellFrom.nix {};
-in shellFrom mypkgs.rhine-gloss
+(import (
+  let
+    lock = builtins.fromJSON (builtins.readFile ./flake.lock);
+  in fetchTarball {
+    url = "https://github.com/edolstra/flake-compat/archive/${lock.nodes.flake-compat.locked.rev}.tar.gz";
+    sha256 = lock.nodes.flake-compat.locked.narHash; }
+) {
+  src =  ./.;
+}).shellNix
```

```suggestion
let
  lock = builtins.fromJSON (builtins.readFile ./flake.lock);
  flake-compat-src = fetchTarball {
    url = "https://github.com/edolstra/flake-compat/archive/${lock.nodes.flake-compat.locked.rev}.tar.gz";
    sha256 = lock.nodes.flake-compat.locked.narHash;
  };
  flake-compat = import flake-compat-src { src =  ./.; };
in
flake-compat.shellNix
```

I find so many big bracket pairs hard to parse